dbms/src/IO/ReadHelpers.h

```diff
@@ -637,28 +652,29 @@ template <typename ReturnType>
 inline ReturnType readDateTimeTextImpl(DateTime64 & datetime64, UInt32 scale, ReadBuffer & buf, const DateLUTImpl & date_lut)
 {
     time_t whole;
-    readDateTimeTextImpl<void>(whole, buf, date_lut);
-    DB::DecimalComponents<DateTime64::NativeType> c{static_cast<DateTime64::NativeType>(whole), 0};
+    if (!readDateTimeTextImpl<bool>(whole, buf, date_lut))
+    {
+        return ReturnType(false);
+    }
+
+    DB::DecimalUtils::DecimalComponents<DateTime64::NativeType> c{static_cast<DateTime64::NativeType>(whole), 0};
 
-    char separator;
-    if (buf.read(separator))
+    if (!buf.eof() && *buf.position() == '.')
     {
-        if (separator != '.')
+        buf.ignore(1); // skip separator
+        const auto count1 = buf.count();
```

Comment:
I sure you can come up with better var name!

Reply:
yep